google-cloud-core-http/src/main/java/com/google/cloud/http/HttpTransportOptions.java

```diff
@@ -163,22 +163,18 @@ public void initialize(HttpRequest httpRequest) throws IOException {
         }
 
         HttpHeaders headers = httpRequest.getHeaders();
-        headers.set("x-goog-api-client", getXGoogApiClientHeader(serviceOptions));
+        headers.set(
+            "x-goog-api-client", getXGoogApiClientHeader(serviceOptions.getLibraryVersion()));
       }
     };
   }
 
-  String getXGoogApiClientHeader(ServiceOptions<?, ?> serviceOptions) {
-    return String.format(
+  String getXGoogApiClientHeader(String libraryVersion) {
+    return String.format(Locale.US,
         "gl-java/%s %s/%s",
-        getJavaVersion(),
+        firstNonNull(Runtime.class.getPackage().getImplementationVersion(), ""),
```

Comment:
I would still prefer to keep this as a helper (even it is a single line) for readability